### ISSUE TITLE
Why are fetures `| undefined` ?

### ISSUE THREAD

**Issue by NullVoxPopuli** [ijlee2/ember-container-query#159]
I'm using ember-container-query v4-beta.1

![image](https://user-images.githubusercontent.com/199018/216779875-5bdc645c-10c8-4914-a47a-15a6bddac0e4.png)

<details><summary>file text here</summary>

```ts
// @ts-ignore
import { hash } from '@ember/helper';

import { ContainerQuery, aspectRatio } from 'ember-container-query';

import constrainVertically from 'limber/modifiers/constrain-vertically';

import type { TOC } from '@ember/component/template-only';

interface Signature {
  Blocks: {
    default: [boolean];
  }
}

export const Orientation: TOC<Signature> =
<template>
  <ContainerQuery
    @features={{hash isVertical=(aspectRatio max=1.2)}}
    {{! grid forces all the contents to take up all available vertical space }}
    class='grid'
    {{constrainVertically}}
    as |query|
  >

    {{#let query.features.isVertical as |isVertical|}}

      {{yield isVertical}}

    {{/let}}

  </ContainerQuery>
</template>;


```

</details>

I would assume that because isVertical is defined and passed to `@features`, that it would always be present in the yielded value?

**Comment by NullVoxPopuli** [comment]
This seems like a reasonable work-around:
```hbs
        {{yield (Boolean isVertical)}}
```
but :shrug:

**Comment by ijlee2** [comment]
I think it's safer (i.e. this addon would depend less on the implementation details of a modifier and Ember's rendering cycle) if we define the type of each query result (which I also called "feature") as `boolean | undefined`.

[Initially, the tracked property `queryResults` is `undefined`](https://github.com/ijlee2/ember-container-query/blob/3.2.0/addon/components/container-query.ts#L34). After the `container-query` modifier does its computations, the tracked property is updated and [takes on the type of `QueryResults`](https://github.com/ijlee2/ember-container-query/blob/3.2.0/addon/components/container-query.ts#L47). In short, there's a brief moment where `isVertical` (in your example) is `undefined`.

I checked that this brief moment exists by creating a helper to log the value:

```ts
/* app/helpers/track-query-results.js */

const TrackQueryResultsHelper = helper((positional) => {
  console.log('Query result: ' + positional[0]);
});

export default TrackQueryResultsHelper;
```

```hbs
<ContainerQuery
  ...
>
  {{track-query-results isVertical}}
</ContainerQuery>
```

<img width="1440" alt="" src="https://user-images.githubusercontent.com/16869656/216963752-fbb08bb3-6ebc-4fdb-a93d-20b9319d8a7a.png">
